pymongo/bulk.py

```diff
@@ -99,8 +101,17 @@ def _merge_command(run, full_result, offset, result):
         upserted = result.get("upserted")
         if upserted:
             n_upserted = len(upserted)
+            rep = codec_options.uuid_representation
             for doc in upserted:
                 doc["index"] = run.index(doc["index"] + offset)
+                # Normalize uuid representation.
+                _id = doc['_id']
+                if rep == UuidRepresentation.UNSPECIFIED:
+                    if isinstance(_id, uuid.UUID):
+                        _id = Binary.from_uuid(_id)
```

Comment:
I couldn't figure out how to trigger this condition

Reply:
This condition will only happen on pymongo 3.